src/btree/bt_npos.c

```diff
@@ -0,0 +1,352 @@
+ * be skipped an eviction pass.
+ *
+ * Eviction wants to be as non-intrusive as possible and never loads pages into memory, while
+ * Partition Cursor can load pages or wait for them to be unlocked. The behavior is controlled by
+ * the flags passed to the functions. The overall set of flags is quite complex.
+ * To simplify the use of this machinery, two helper functions are provided:
+ *   - __wt_page_from_npos_for_eviction
+ *   - __wt_page_from_npos_for_read
+ *
+ *    === Detailed description.
+ *
+ * Normalized position is a number in the range of 0 .. 1 defining a page's position in the tree.
+ * In fact, each page occupies a range of positions. For example, if there are 5 pages then
+ * positions of pages are:
+ *   - [0.0 .. 0.2) -> page 0
+ *   - [0.2 .. 0.4) -> page 1
+ *   - [0.4 .. 0.6) -> page 2
+ *   - [0.6 .. 0.8) -> page 3
+ *   - [0.8 .. 1.0] -> page 4
+ * The starting point is inclusive, the ending point is exclusive.
+ *
+ * When calculating a page's position, the returned result is always in the range of 0 .. 1. Because
+ * of that, any number outside of this range can be used as an invalid position when storing it.
+ *
+ * When retrieving a page, any number below 0 will lead to the first page, any number above 1 will
+ * lead to the last page. This has useful consequences discussed below.
+ *
+ *    === Finding a page from its normalized position.
+ *
+ * If all leaf pages are attached straight to the root, then finding a page from its normalized
+ * position is just as simple as multiplying it by the number of leaf pages and using the integer
+ * part of the result as the page's index.
+ *
+ * If there are multiple levels, then the process is similar: the integer part is used as an index
+ * at the current level, and the fractional part is used as a normalized position at the next level.
+ *
+ * This process is repeated until we reach a leaf page.
+ *
+ * The remaining fractional part at the leaf page can be potentially used to find an exact key
+ * on the page. This is not implemented since there's no need for it.
+ *
+ *    === Calculating a page's normalized position.
+ *
+ * As opposed to finding a page from its normalized position, the process goes back from the page
+ * up to the root. The process is a reverse of the finding process.
+ *
+ * A nuance is that because there is a whole range of numbers corresponding to a page, the user
+ * can choose a starting position within the page. Say, numbers closer to 0 will point to somewhere
+ * closer to the beginning of the page, and numbers closer to 1 will point close to the end.
+ *
+ * If there's only one level, then the normalized position is just the page's index (plus fractional
+ * starting point) divided by the number of pages at the parent level.
+ *
+ * If there are multiple levels, the process is repeated until we reach the root with starting
+ * point being whatever has been calculated at the previous level.
+ *
```

Comment:
I would add an example calculation here:
```
Suppose we have a depth 2 tree, at the lowest level we are the 6th out of 10 pages and at the higher level the 3rd out of 5 pages. Furthermore we want to start around quarter of the way into the page.

Initially we have a starting position of 0.25, our page is at the 6/10th position in this level of the tree. This gives us the following calculation:
Level 2 position = (6 + 0.25)/10  = 0.625

Then we add the position from level 1:
Final position = (3 + Level 2 position)/5 = (3 + 0.625)/5 = 0.725.
```

I think you may need to workshop my comment format a bit.